Model/Api/UpdateCart.php

```diff
@@ -203,6 +203,21 @@ public function execute($cart, $add_items = null, $remove_items = null, $discoun
                         // Already sent a response with error, so just return.
                         return false;
                     }
+                    // Adding the same data into immutable quote
+                    // in order to avoid missing addons in parent quote
+                    $addToImmutableQuote = true;
```

Comment:
I just realized - we need to remove from immutable quote as well.
See line 259 of the same file